test/e2e/bundle-pages-router-dependencies/test/index.test.ts

```diff
@@ -0,0 +1,34 @@
+/* eslint-env jest */
+
+import { join } from 'path'
+import { nextTestSetup } from 'e2e-utils'
+
+describe('bundlePagesRouterDependencies', () => {
+  const { next } = nextTestSetup({
+    files: join(__dirname, '..'),
+  })
+
+  it('should render the page', async () => {
+    const html = await next.render('/')
+    expect(html).toInclude('bar and baz')
+  })
+
+  it('should bundle pages dependencies except for serverExternalPackages', async () => {
+    const html = await next.render('/')
+    expect(html).toMatch(
+      /require\(&[\w#]+;.\/opted-out-external-package-source&[\w#]+;\)/
+    )
+    expect(html).not.toMatch(
+      /require\(&[\w#]+;.\/external-package-source&[\w#]+;\)/
+    )
+    if (process.env.TURBOPACK) {
+      expect(html).toInclude(
+        'node_modules/external-package/external-package-source.js [ssr] (ecmascript)'
+      )
+    } else {
+      expect(html).toInclude(
+        '/*! ./external-package-source */ &quot;./node_modules/external-package/external-package-source.js&quot;'
+      )
+    }
```

Comment:
In production this probably end up being different.

Reply:
Yeah, the require becomes `r(481)`.
I've removed these last two assertions now. Or is there some other way to assert this?